### PR TITLE
Support version upgrade for MSI packages

### PR DESCRIPTION
**Pull request by Kausik-A**
**Description:** 

This PR super-seeds #1764. Two major changes are defined in the PR

1. The new functionality updates the `version` attribute inside [tools/packaging/windows/aws-otel-collector.wxs](https://github.com/aws-observability/aws-otel-collector/compare/main...Kausik-A:newapproach?expand=1#diff-c8bf1b245b317bbe0943e601c014b19bee21d8d5dcfab8e1a88c70538699509e) when [bump-version-and-create-release-note.sh](https://github.com/aws-observability/aws-otel-collector/compare/main...Kausik-A:newapproach?expand=1#diff-b1c7c6870407646abec13c8b0a0f806344c0d01f47e9a1db7ec379a97b17f912) is run as part of CD process. This will ensure we build windows packages with the right version

2. According to wix documentation, it is advised to have a static product ID when building packages so introduced a unique string to update this.

3. Add `MajorUpgrade` tag to ensure new version of collector uninstalls the older version of ADOT collector and installs the new one. This mitigates the issue of two collectors being installed and run simultaneously.

**Note-1**: Currently the sed patterm supports [0-9].[0-9].[0-9]* version pattern. If we ever decide to release with a full semantic version with prefix as  `vX.XX.XX-prerelease` then the `.wxs` file will need to be **manually updated**

Note-2: We need not update the CD workflow as mentioned in issue #1765. We update the [aws-otel-collector.wxs](https://github.com/aws-observability/aws-otel-collector/compare/main...Kausik-A:newapproach?expand=1#diff-c8bf1b245b317bbe0943e601c014b19bee21d8d5dcfab8e1a88c70538699509e) with value inside [VERSIONS](https://github.com/aws-observability/aws-otel-collector/blob/main/VERSION). So just checking the `VERSION` with the version inputted in the `CD workflow dispatch` would be enough to ensure an incorrect version of windows is not released. This is already been done [here](https://github.com/aws-observability/aws-otel-collector/blob/4224c18fefdcaf40dc0cd6555258df01645b49ed/.github/workflows/CD.yml#LL89C16-L89C16)

**Testing:** 

Tested locally buy building different versions of ADOT windows packages and installing them on a local machine

Will add screenshots in a while

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tools/packaging/windows/aws-otel-collector.wxs [hunk 1]
```diff
@@ -20,6 +22,12 @@ SummaryCodepage='1252'
 InstallScope="perMachine"
 Platform="x64"
     />
+
+    <MajorUpgrade
+    AllowDowngrades="no" 
+    DowngradeErrorMessage="A later version of ADOT Collector is already installed. Setup will now exit"
+    AllowSameVersionUpgrades="no" />
```
Comment (bryan-aguilar) [inline tools/packaging/windows/aws-otel-collector.wxs:29]: Are these default values other than the `DowngradeErrorMessage`?

File: tools/packaging/windows/aws-otel-collector.wxs [hunk 2]
```diff
@@ -1,13 +1,15 @@
 <?xml version='1.0' encoding='UTF-8'?>
 <Wix xmlns="http://schemas.microsoft.com/wix/2006/wi">
-    <Product Id="*"
+    <Product Id="82568046-50CE-FCDE-9645-518945DACEFC"
```
Comment (bryan-aguilar) [inline tools/packaging/windows/aws-otel-collector.wxs:3]: What was used to generate this GUID?

File: tools/packaging/windows/aws-otel-collector.wxs (hunk 1 above)
Comment (Kausik-A) [reply tools/packaging/windows/aws-otel-collector.wxs:29]: Both the remaining parameters defaults to no

File: tools/packaging/windows/aws-otel-collector.wxs (hunk 2 above)
Comment (Kausik-A) [reply tools/packaging/windows/aws-otel-collector.wxs:3]: Used a random generator to match the regex pattern specified by Wix

Comment (bryan-aguilar) [reply tools/packaging/windows/aws-otel-collector.wxs:3]: Can you reference this guide please? https://wixtoolset.org/docs/v3/howtos/general/generate_guids/
We may want to use https://www.guidgen.com/

I'm not sure if the random number generator meets the requirements.

Comment (Kausik-A) [reply tools/packaging/windows/aws-otel-collector.wxs]: Updated it using https://www.guidgen.com/

File: tools/packaging/windows/aws-otel-collector.wxs [hunk 3]
```diff
@@ -1,13 +1,15 @@
 <?xml version='1.0' encoding='UTF-8'?>
 <Wix xmlns="http://schemas.microsoft.com/wix/2006/wi">
-    <Product Id="*"
+    <Product Id="ebbd8548-75d1-41d3-a402-abe189f0c167"
```
Comment (bryan-aguilar) [inline tools/packaging/windows/aws-otel-collector.wxs:3]:
```suggestion
    <Product Id="EBBD8548-75D1-41D3-A402-ABE189F0C167"
```

**Review by bryan-aguilar — APPROVED** [review]

File: tools/release/bump-version-and-create-release-note.sh [hunk 4]
```diff
@@ -35,6 +35,9 @@ docker run --rm -it -v "$(pwd)":/usr/local/src/your-app ferrarimarco/github-chan
 # bump the version
 echo "${RELEASE_VERSION}" >VERSION
 
+#update aws-otel-collector msi file
+sed -i '' "s/\(<Product [^>]*Version=\"\)[^\"]*\"/\1${RELEASE_VERSION:1}\"/" tools/packaging/windows/aws-otel-collector.wxs
```
Comment (bryan-aguilar) [inline tools/release/bump-version-and-create-release-note.sh:39]: Can you post test results for this oneliner?

**Review by bryan-aguilar — CHANGES_REQUESTED** [review]

File: tools/release/bump-version-and-create-release-note.sh (hunk 4 above)
Comment (bryan-aguilar) [reply tools/release/bump-version-and-create-release-note.sh:39]: I tried running it locally but the file was not updated
```
RELEASE_VERSION=0.26.0 sed -i '' "s/\(<Product [^>]*Version=\"\)[^\"]*\"/\1${RELEASE_VERSION:1}\"/" tools/packaging/windows/aws-otel-collector.wxs
```

Comment (bryan-aguilar) [reply tools/release/bump-version-and-create-release-note.sh:39]: Also, does `RELEASE_VERSION` usually include a `v` prefix? Do we need to remove that before doing the search and replace?

Comment (bryan-aguilar) [reply tools/release/bump-version-and-create-release-note.sh:39]: Also, please add a comment saying that this only supports `[0-9]*.[0-9]*.[0-9]*` version pattern. If we decide to release with a full sem ver `vX.XX.XX-prerelease` then the `.wxs` file will need to be manually updated.

Comment (Kausik-A) [reply tools/release/bump-version-and-create-release-note.sh]: > I tried running it locally but the file was not updated

I have updated the sed pattern. Here are the results

### .wxs file before
Screen Shot 2023-02-01 at 4 51 58 PM

### .wxs file after (as expected)
Screen Shot 2023-02-01 at 4 53 16 PM

Comment (Kausik-A) [reply tools/release/bump-version-and-create-release-note.sh]: > Also, does RELEASE_VERSION usually include a v prefix? Do we need to remove that before doing the search and replace

the RELEASE_VERSION has 'v' prefix it is but this first character is removed by `${RELEASE_VERSION:1}`

> Also, please add a comment saying that this only supports [0-9]*.[0-9]*.[0-9]* version pattern. If we decide to release with a full sem ver vX.XX.XX-prerelease then the .wxs file will need to be manually updated.

Added it in PR description along with comment inside the shell script

Comment (bryan-aguilar) [reply tools/release/bump-version-and-create-release-note.sh]: `RELEASE_VERSION=v0.26.0 sed -i '' "s/^Version\=\"[0-9]*.[0-9]*.[0-9]*\"/Version=\"${RELEASE_VERSION:1}\"/" tools/packaging/windows/aws-otel-collector.wxs`
Results in `Version=".26.0"`  when I run it locally.

Comment (bryan-aguilar) [reply tools/release/bump-version-and-create-release-note.sh]: Disregard above. This was a local environment issue. I can confirm it works.

**Comment by bryan-aguilar** [comment]
Closing in favor of #1808

**Pull request closed without merging**